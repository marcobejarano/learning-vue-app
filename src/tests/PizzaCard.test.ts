import { shallowMount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import PizzaCard from "@/components/PizzaCard.vue";

describe('PizzaCard', () => {
  it('should render the pizza details', () => {
    const pizza = {
      id: "1",
      title: 'Test Pizza',
      description: 'Test Pizza Description',
      image: 'test-pizza.jpg',
      price: "10",
      quantity: 10,
    };

    const wrapper = shallowMount(PizzaCard, {
      props: {
        pizza,
      },
    });

    expect(wrapper.find('.pizza--details-wrapper')).toBeTruthy();

    expect(
      wrapper.find('.pizza--inventory-stock').text()
    ).toBe(`Stock: ${pizza.quantity}`);

    expect(wrapper.find('.pizza--inventory-price').text()).toBe(`$ ${pizza.price}`);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render the pizza image and alt text', () => {
    const pizza = {
      id: "1",
      title: 'Test Pizza',
      description: 'Test Pizza Description',
      image: 'test-pizza.jpg',
      price: "10",
      quantity: 10,
    };

    const wrapper = shallowMount(PizzaCard, {
      props: {
        pizza,
      },
    });

    const img = wrapper.find('img');

    expect(img.attributes().alt).toEqual(pizza.title);
    expect(img.attributes().src).toEqual(pizza.image);
  });

  it('should emit add-to-cart event when add to cart button is clicked', () => {
    const pizza = {
      id: "1",
      title: 'Test Pizza',
      description: 'Test Pizza Description',
      image: 'test-pizza.jpg',
      price: "10",
      quantity: 10,
    };

    const wrapper = shallowMount(PizzaCard, {
      props: {
        pizza,
      },
    });

    const button = wrapper.find('button');
    button.trigger('click');

    expect(wrapper.emitted()['add-to-cart']).toBeTruthy();
    expect(wrapper.emitted()['add-to-cart'][0]).toEqual([
      { id: pizza.id, quantity: 1 }
    ]);
  })
});
