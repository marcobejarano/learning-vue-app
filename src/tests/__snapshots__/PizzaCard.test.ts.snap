// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PizzaCard > should render the pizza details 1`] = `
"<article data-v-7f67d26c="" class="pizza--container"><img data-v-7f67d26c="" src="test-pizza.jpg" alt="Test Pizza" height="200" width="300">
  <p data-v-7f67d26c="">Test Pizza Description</p>
  <div data-v-7f67d26c="" class="pizza--inventory">
    <div data-v-7f67d26c="" class="pizza--inventory-stock">Stock: 10</div>
    <div data-v-7f67d26c="" class="pizza--inventory-price">$ 10</div><!-- <span v-if="isInCart">In cart</span> -->
  </div><button data-v-7f67d26c="">Add to cart</button><!-- <button class="pizza--add" @click="addToCart">Add to cart</button> -->
</article>"
`;
